File: source/yeon/6주차/code/4796캠핑.cpp
// 백준 4796 캠핑
// cpp sovled by Greedy algorithm

#include <iostream>
int main(void){
    int L,P,V, cnt =1;
    do{
        scanf("%d %d %d", &L, &P, &V);
        if(L == 0 && P == 0 && V == 0) break;
        else{
            if(V % P < L)
                printf("Case %d: %d\n", cnt, (V/P)*L + V%P);
            else
                printf("Case %d: %d\n", cnt, (V/P)*L + L);
            cnt ++;
        }
    }while(L != 0 && P != 0 && V != 0);
    return 0;
}

File: source/yeon/6주차/code/4796캠핑.py
# 백준 4796 캠핑
# python sovled by Greedy algorithm
arr = []
L,P,V = 1,1,1

# get input from user
while True:
    L,P,V = map(int,input().split())
    arr.append([L,P,V])
    if L == 0 and P == 0 and V == 0 : break

for i in range(len(arr)-1):
    # V % P < L
    if arr[i][2] % arr[i][1] < arr[i][0] : 
        print("Case {}: {}".format(i+1, int(arr[i][2]/arr[i][1])*arr[i][0] + arr[i][2] % arr[i][1]))
    # V % P > L 
    else :
        print("Case {}: {}".format(i+1, int(arr[i][2]/arr[i][1])*arr[i][0] + arr[i][0]))
